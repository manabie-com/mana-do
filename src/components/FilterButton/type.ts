export interface FilterButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  active?: boolean;
}