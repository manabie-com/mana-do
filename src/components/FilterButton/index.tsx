import React from "react";

export interface FilterButtonProps {
  active?: boolean;
  onClick: () => void;
  children: string;
}

const FilterButton: React.FC<FilterButtonProps> = ({
  active = false,
  onClick,
  children,
}) => {
  return (
    <button
      data-testid="btn-filter"
      className={`Action__btn Filter__btn ${active ? "active" : ""}`}
      onClick={() => onClick()}
    >
      {children}
    </button>
  );
};

export default FilterButton;
