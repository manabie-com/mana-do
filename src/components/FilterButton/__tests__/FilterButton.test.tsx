import React from "react";
import { expect, test, describe } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import FilterButton, { FilterButtonProps } from "../index";

describe("FilterButton Snapshot", () => {
  test("should render with active status", () => {
    const props: FilterButtonProps = {
      active: true,
      onClick: () => {},
      children: "TEST",
    };
    const { container } = render(<FilterButton {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render without active status", () => {
    const props: FilterButtonProps = {
      active: false,
      onClick: () => {},
      children: "TEST",
    };
    const { container } = render(<FilterButton {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

test("should render a button with correct children", () => {
  const text = "TEST";
  const props: FilterButtonProps = {
    active: false,
    onClick: () => {},
    children: text,
  };
  const element = render(<FilterButton {...props} />);
  const filterElement = element.getByTestId("btn-filter");
  expect(filterElement.innerHTML).toEqual(text);
});
