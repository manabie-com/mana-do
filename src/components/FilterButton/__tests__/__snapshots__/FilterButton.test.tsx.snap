// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton Snapshot should render with active status 1`] = `
<button
  class="Action__btn Filter__btn active"
  data-testid="btn-filter"
>
  TEST
</button>
`;

exports[`FilterButton Snapshot should render without active status 1`] = `
<button
  class="Action__btn Filter__btn "
  data-testid="btn-filter"
>
  TEST
</button>
`;
